src/Editor.ts

```diff
@@ -431,12 +455,20 @@ function getActiveFormats(editor: Editor, doc: TextDocument, selection = doc.sel
   const at = normalizeRange(selection)[0];
   // If start of a non-empty line, use the format of the first character, otherwise use the format of the preceeding
   let formatAt = at;
+  let formatTo = at + 1;
   const attributes = doc.getTextFormat(formatAt);
+  const nextAttributes = doc.getTextFormat(formatTo);
   const format: AttributeMap = {};
   // Sort them by the order found in marks and be efficient
   Object.keys(attributes).forEach(name => {
     const type = formats.get(name);
-    if (type && (type.greedy !== false || doc.getTextFormat(at)[name])) {
+    if (type && type.greedy !== false) {
+      format[name] = attributes[name];
+    }
+  });
+  Object.keys(nextAttributes).forEach(name => {
```

Comment:
This fixes the problem of typing into the middle of a link.

Reply:
Yes, I think this is what was needed. Nice.